tests/check.sh

```diff
@@ -3,6 +3,27 @@ set -euo pipefail
 dn=$(dirname "$0")
 srcdir=$(cd "${dn}"/.. && pwd)/src
 
+# short utility function to do version comparison
+# hat-tip to StackOverflow - https://stackoverflow.com/a/24067243
+function version_gt() {
+  test "$(printf '%s\n' "$@" | sort -V | head -n 1)" != "$1";
+}
+
+# Check for ShellCheck version
+# We want to make use of the `-x` flag, which was made available
+# in v0.4.0 (see CHANGELOG.md - https://github.com/koalaman/shellcheck/blob/master/CHANGELOG.md)
+# and earlier versions choke on the shellcheck directives, so we have to
+# disable its usage entirely
+ver=$(shellcheck --version | grep version: | awk '{print $2}')
+set +e
+version_gt "$ver" "0.4.0"
```

Comment:
Not as pretty, but could also just do `shellcheck -h |& grep -- --external-sources`.

Reply:
wow.. TIL about `|&` been using `2>&1 |` for way too long